c/datasetjson.c

```diff
@@ -2251,9 +2254,238 @@ void respondWithHLQNames(HttpResponse *response, MetadataQueryCache *metadataQue
+      strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+    }
+  }
+
+  /* Initialize number of generations */
+  if (request->numGenerations != 0) {
+    snprintf(field, sizeof(field)-1, "MAXGENS(%d) ", request->numGenerations);
+    strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+  }
+ 
+  /* Initialize volumn if defined */
+  if (request->volume != NULL) {
+    snprintf(field, sizeof(field)-1, "VOL(%s) ", request->volume);
+    strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+  }
+
+  /* Initialize dataClass if defined */
+  if (request->dataClass != NULL) {
+    snprintf(field, sizeof(field)-1, "DATACLAS(%s) ", request->dataClass);
+    strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+  }
+
+  /* Initialize management class if defined */
+  if (request->manageClass != NULL) {
+    snprintf(field, sizeof(field)-1, "MGMTCLAS(%s) ", request->manageClass);
+    strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+  }
+
+  /* Initialize management class if defined */
+  if (request->averageRecord != NULL) {
+    //snprintf(field, sizeof(field)-1, "BLOCK(%s) ", request->averageRecord);
+    snprintf(field, sizeof(field)-1, "AVGREC(%s) ", request->averageRecord);
+    strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+  }
+
+  /* Initialize expiration data  */
+  if (request->expiration != NULL) {
+    snprintf(field, sizeof(field)-1, "EXPDL(%s) ", request->expiration);
+    strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+  }
+
+  /* Initialize expiration data  */
+  if (request->spaceUnit != NULL) {
+    snprintf(field, sizeof(field)-1, "SPACE(%s) ", request->spaceUnit);
+    strncat( below2G->request, field,
+             sizeof (below2G->request) - strlen(below2G->request));
+  }
+
+  /****************************/
+  /* Create the qsam data set */
+  /****************************/
+  printf (" REQUEST: %s\n",  below2G->request);
+  status = bpxwdyn(below2G->request);
+
+  /* If error, don't try to free, decipher error code */
+  if (status != 0) {
+    int cause = 0, reasonIndex;
+    char *reason;
+
+    /* status between -20 and -9999 represents Key errors. */
+    /* The key index is (-status -20)                      */
+    /* status below -10000 has the IEFDB476 return code in */
+    /* lower 2 digits (decimal) of the status.             */
+    /* Status code between  -1610612737 and -2147483648    */
+    /* contains the S99ERROR code.                         */
+    reason = strtok(below2G->request, " ");
+    sprintf(field,"", "");  
+    if ((status <= -20) && (status >= -9999)) {
+      cause = (-status - 20) %100;
```

Comment:
What are all these magic numbers? The comments are not enough, please use `#define`s.

Reply:
Using #defines will not make the code any more understandable then the comments.  
Have modified.

Reply:
I think the comments are great but I agree with @ifakhrutdinov that constants would make the code more readable. I would use both.